spring-cloud-gcp-core/pom.xml

```diff
@@ -22,5 +22,9 @@
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-core</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-autoconfigure</artifactId>
```

Comment:
Cyclic dependency?
Why should we worry about autoconfig here if it is just `core`?
Not sure what I am missing, but would like to take your attention on this.

Thanks

Reply:
`spring-boot-autoconfigure`, not `spring-cloud-gcp-autoconfigure`.

We need `spring-boot-autoconfigure` to get `GcpProperties`'s `@ConfigurationProperties` annotation. LMK if I should be using another dependency.

Reply:
Oh! My bad. I see what is that.
Well, I think the problem that all the `@ConfigurationProperties` must be as a part of our `autoconfigure` not more.

Not sure how have you arranged it with Stephane, but I feel like something in architecture is still missing. Maybe there is still something in the target modules what is Boot-based? So, that should be moved to the `autoconfigure` module as well...

Reply:
[Other spring-boot projects](https://github.com/spring-projects/spring-boot/tree/master/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/health) seem to be keeping their `@ConfigurationProperties` in the autoconfigure module as well.

I reviewed our project and all `@ConfigurationProperties` are used only within autoconfigure, except for the config and SQL ones.

For the config one, we might be able to make `GoogleConfigPropertySourceLocator` take in individual arguments instead of `GcpConfigProperties` (e.g., `config name`, `enabled`, etc.).
The SQL properties are used by the `JdbcUrlProviders`, but I think we can add those to the autoconfigure module as well..

Reply:
> For the config one, we might be able to make 

Are you going to do those changes to avoid this `spring-boot-autoconfigure` dependency in the `core`?
Or do you mean to do that in the separate issue?

Thanks